typing/env.ml

```diff
@@ -2002,9 +2002,10 @@ let add_components slot root env0 comps =
   }
 
 let open_signature slot root env0 =
-  match get_components (find_module_components root env0) with
-  | Functor_comps _ -> None
-  | Structure_comps comps ->
+  match get_components_res (find_module_components root env0) with
+  | Error _
+  | Ok (Functor_comps _) -> None
```

Comment:
`open_pers_signature` right below seems to assume that the only `None` case here is a functor. What should we do now that this assumption is not correct anymore? I am also nervous about the `Env_open` cases in envaux.ml which also `assert false` on a `None` result.

Could maybe change the return type from an `option` to something more informative, so that the `Error` cases can be distinguished from the `Functor_comps` case?

`env_from_summary` seems to expect that `open_signature` could also raise the exception `Not_found`. Is this still the case today?

Reply:
Indeed, lots of stuff broken.
Switch everything to Error + polymorphic variant ?

Reply:
> `env_from_summary` seems to expect that open_signature could also raise the exception Not_found. Is this still the case today?

Callers of `open_pers_signature` make the same assumption. And yes, it can happen: the call to `find_module_components` can raise `Not_found`.

Also, it seems that `open_pers_signature` is only called from `Matching`, on a module which we know not to be an alias.

I have no comment about `env_from_summary`, I guess one would need to compile some files, with one aliasing the other, then try running cmt2annot (or the debugger) after having removed the cmi of the first one?

Reply:
Ok, turning into a nightmare.
I switched everything to Error.
Now `open_signature` returns ` (Env.t, [``Not_found | ``Functor]) result`, and both the exception `Not_found` and errors in `get_components` are mapped to `Error ``Not_found`